tools/packaging/kata-deploy-cc/scripts/kata-deploy.sh

```diff
@@ -211,9 +211,11 @@ function configure_containerd_runtime() {
 	fi
 	local runtime_table="plugins.${pluginid}.containerd.runtimes.$runtime"
 	local runtime_type="io.containerd.$runtime.v2"
-	local cri_handler_value=""
-        if echo "${runtime_type}" | grep -q -v -e "kata-remote\.v2" -e "kata\.v2"; then
-                cri_handler_value="cc"
```

Comment:
I think these lines were copied from the `cloud-api-adaptor` repo.
https://github.com/confidential-containers/cloud-api-adaptor/blob/2536ae8ef62e5aed5cda224a428d389e52b68768/install/runtime-payload/scripts/kata-deploy.sh#L187-L189

The intention of this code logic is that `cri_handle` is set to `"cc"` unless shim is `remote` or the default one. The default shim is `remote` in `kata-deploy.sh` in cloud-api-adaptor is `remote`, so I think this logic is valid there.
https://github.com/confidential-containers/cloud-api-adaptor/blob/2536ae8ef62e5aed5cda224a428d389e52b68768/install/runtime-payload/scripts/kata-deploy.sh#L25

However, the default shim is `qemu` in this kata-containers repo, so I think this logic should be fixed.
https://github.com/kata-containers/kata-containers/blob/4ab93e662c5a0d4a06afe1862d934ee8eec1e966/tools/packaging/kata-deploy-cc/scripts/kata-deploy.sh#L27

@bpradipt @stevenhorsman Is this correct?

Reply:
Hey Yohei,
Just to confirm - the current behaviour results in the default `kata` and `remote` missing the cc cri_handler, but qemu doesn't:
```
[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata]
  cri_handler = ""
  runtime_type = "io.containerd.kata.v2"
  privileged_without_host_devices = true
  pod_annotations = ["io.katacontainers.*"]
  [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata.options]
    ConfigPath = "/opt/confidential-containers/share/defaults/kata-containers/configuration.toml"
[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata-remote]
  cri_handler = ""
  runtime_type = "io.containerd.kata-remote.v2"
  privileged_without_host_devices = true
  pod_annotations = ["io.katacontainers.*"]
  [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata-remote.options]
    ConfigPath = "/opt/confidential-containers/share/defaults/kata-containers/configuration-remote.toml"
[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata-qemu]
  cri_handler = "cc"
  runtime_type = "io.containerd.kata-qemu.v2"
  privileged_without_host_devices = true
  pod_annotations = ["io.katacontainers.*"]
  [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata-qemu.options]
    ConfigPath = "/opt/confidential-containers/share/defaults/kata-containers/configuration-qemu.toml"
```

so I'm not sure if it completely broken (if so then 0.5 release would be?), but I think your improvement is better!

Reply:
Right. My patch will change the `cri_handler` value of the default one to `"cc"`. This is what I'd like to confirm.

I think  the `cri_handler` value of the default one was also set to `"cc"` before this commit.
https://github.com/kata-containers/kata-containers/commit/6a7c39824a5f91804cbc954252a9cc7a86743790#diff-d6bd3f1d7902606e43fc151a5a9a3a19a3624b45a08e0e7fcdb9385e272578f5R201-R204

I guess everyone uses specific names like `kata-qemu`, and no one uses the default name `kata`, so no one complains about this.

Reply:
> Right. My patch will change the cri_handler value of the default one to "cc". This is what I'd like to confirm.

My understanding is that this is correct and the desired behaviour